src/Pact/Types/Command.hs

```diff
@@ -196,45 +270,56 @@ verifyCommand orig@Command{..} =
                     =<< A.eitherDecodeStrict' _cmdPayload
 
     verifiedHash = verifyHash _cmdHash _cmdPayload
-{-# INLINE verifyCommand #-}
 
 hasInvalidSigs :: PactHash -> [UserSig] -> [Signer] -> Maybe String
 hasInvalidSigs hsh sigs signers
   | not (length sigs == length signers)  = Just "Number of sig(s) does not match number of signer(s)"
-  | otherwise                            = if (length failedSigs == 0)
-                                           then Nothing else formatIssues
-  where verifyFailed (sig, signer) = not $ verifyUserSig hsh sig signer
-
-        -- assumes nth Signer is responsible for the nth UserSig
-        failedSigs = filter verifyFailed (zip sigs signers)
-        formatIssues = Just $ "Invalid sig(s) found: " ++ show (J.encode . J.Array <$> failedSigs)
-
-
-verifyUserSig :: PactHash -> UserSig -> Signer -> Bool
-verifyUserSig msg UserSig{..} Signer{..} =
-  case (pubT, sigT, addrT) of
-    (Right p, sig, addr) ->
-      (isValidAddr addr p) &&
-      verify (toScheme $ fromMaybe defPPKScheme _siScheme)
-             (toUntypedHash msg) (PubBS p) sig
-    _ -> False
-  where pubT = parseB16TextOnly _siPubKey
-        sigT = case parseB16TextOnly _usSig of
-          Left _ -> SigBS (Text.encodeUtf8 _usSig)
-          Right bs -> SigBS bs
-        addrT = parseB16TextOnly <$> _siAddress
-        isValidAddr addrM pubBS = case addrM of
-          Nothing -> True
-          Just (Left _) -> False
-          -- All current cases of `_siScheme` require the same relationship
-          -- between `pubBS` and `givenAddr`. But we enumerate them anyway,
-          -- so that if another scheme is added in the future, this use site
-          -- will warn us that we need to consider the `pubBS`/`givenAddr`
-          -- relationship for that scheme, since it may be different.
-          Just (Right givenAddr) -> case _siScheme of
-            Nothing -> pubBS == givenAddr
-            Just ED25519 -> pubBS == givenAddr
-            Just WebAuthn -> pubBS == givenAddr
+  | otherwise                            = verifyUserSigs hsh (zip sigs signers)
+
+verifyUserSigs :: PactHash -> [(UserSig, Signer)] -> Maybe String
+verifyUserSigs hsh sigsAndSigners =
+  if not (null failedSigs)
+  then formatIssues
+  else Nothing
+  where
+  getFailedVerify (sig, signer) =
+    [ over each Text.pack (show signer, err) | Left err <- [verifyUserSig hsh sig signer] ]
```

Comment:
Isn't this just `(Text.pack $ show signer, Text.pack err)`? why not just do it explicitly instead of `lens` golfing?